src/openms/source/CONCEPT/QCMBRalignment.cpp

```diff
@@ -11,90 +11,165 @@ QCMBRalignment::~QCMBRalignment()
 
 }
 
-int QCMBRalignment::MBRAlignment(MetricMap& raws, MetricMap& outSeq, MetricMap& outCorrRT, MetricMap& outOrigRT, MetricMap& outSpectra) const
+int QCMBRalignment::MBRAlignment(MzTab& mztab) const
 {
-      vector<String> rawFiles;
-  	  vector<String> sequences;
-  	  vector<float> originalRTs;
-  	  vector<float> correctedRTs;
-  	  vector<String> spectra;
+  vector<String> rawFiles;
+  vector<String> sequences;
+  vector<float> originalRTs;
+  vector<float> correctedRTs;
+  vector<String> spectra;
   	  
-  	  vector<FeatureMap> maps;
+  vector<FeatureMap> maps;
   	  
-  	  for(vector<pair<String,FeatureMap>>::const_iterator it = FeatMap_.begin();it!=FeatMap_.end();++it){
-        if(it->first=="MapRTTransformer"){maps.push_back(it->second);}
-      }
+  for(vector<pair<String,FeatureMap>>::const_iterator it = feat_map_.begin();it!=feat_map_.end();++it)
+  {
+    if(it->first=="MapRTTransformer"){maps.push_back(it->second);}
+    }
       
-    
       for (vector<FeatureMap>::const_iterator m_it = maps.begin(); m_it!=maps.end();m_it++) {
+    		    		
+        String rfile = m_it->getMetaValue("spectra_data");	
+    	for (vector<Feature>::const_iterator f_it = m_it->begin(); f_it!=m_it->end();f_it++) 
+    	{
     		
-    		//cout << m_it->getMetaValue("spectra_data") << endl;
-    		
-    		String rfile = m_it->getMetaValue("spectra_data");
-    		
-    		for (vector<Feature>::const_iterator f_it = m_it->begin(); f_it!=m_it->end();f_it++) {
-    		
- 				vector<PeptideIdentification> pep_id = f_it->getPeptideIdentifications();
+ 		  vector<PeptideIdentification> pep_id = f_it->getPeptideIdentifications();	
+ 		  if (pep_id.empty()) 
+ 		  {
  				
- 				if (pep_id.empty()) {
- 					//Nan	
- 					rawFiles.push_back(rfile);
- 					cout << rfile << endl;
- 					float corrRT = f_it->getRT();
- 					float orRT = f_it->getMetaValue("original_RT");
- 					correctedRTs.push_back (corrRT);
- 					cout << corrRT << endl;
- 					originalRTs.push_back (orRT);
- 					cout << orRT << endl;
- 					sequences.push_back ("NaN");
- 					cout << "NaN" << endl;
- 					spectra.push_back ("0");
- 					cout << 0 << endl;
- 				}
+ 			rawFiles.push_back(rfile);
+ 			float corrRT = f_it->getRT();
+ 			float orRT = f_it->getMetaValue("original_RT");
+ 			correctedRTs.push_back (corrRT);
+ 			originalRTs.push_back (orRT);
+ 			sequences.push_back ("NaN");
+ 			spectra.push_back ("0");
+ 		  }
  				
- 				else {
- 					// pep_id and sequence + spectrum ident
- 					for (vector<PeptideIdentification>::iterator p_it = pep_id.begin(); p_it!=pep_id.end();p_it++) {
-    					//cout << "corrected_RT: " << p_it->getRT()  << endl;
-    					//cout << "original_RT: " << p_it->getMetaValue("original_RT") << endl;
+ 		  else 
+ 		  {
+ 		    // pep_id and sequence + spectrum ident
+ 			for (vector<PeptideIdentification>::iterator p_it = pep_id.begin(); p_it!=pep_id.end();p_it++) 
```

Comment:
iterating over all peptideIDs is probably not a good idea. The first ID and the first hit should be taken (if present).
Otherwise, the number of mzTab rows depends on the number of candidates a search engine generates.
It should depend on the number of features though.